elephant/spectral.py

```diff
@@ -242,6 +242,121 @@ def welch_psd(signal, n_segments=8, len_segment=None,
     return freqs, psd
 
 
+def multitaper_psd(signal, fs=1, nw=4, num_tapers=None,
+                   peak_resolution=None):
+    """
+    Estimates power spectrum density (PSD) of a given 'neo.AnalogSignal'
+    using Multitaper method
+
+    The PSD is obtained through the following steps:
+
+    1. Calculate 'num_tapers' approximately independent estimates of the
+       spectrum by multiplying the signal with the discrete prolate spheroidal
+       functions (also known as Slepian function) and calculate the PSD of the
+       products
+
+    2. Average the approximately independent estimates to decrease overall
+       variance of the estimates
+
+    Parameters
+    ----------
+    signal : neo.AnalogSignal
+        Time series data of which PSD is estimated. When `signal` is np.ndarray
+        sampling frequency should be given through keyword argument `fs`.
+    fs : float, optional
+        Specifies the sampling frequency of the input time series
+        Default: 1.0
+    nw : float, optional
+        Time bandwidth product
+        Default: 4.0
+    num_tapers : int, optional
+        Number of tapers used in 1. to obtain estimate of PSD. By default
+        [2*nw] - 1 is chosen.
+        Default: None
+    peak_resolution : float, optional
+        Desired frequency resolution of the obtained PSD estimate. When given
+        as a `float`, it is taken as frequency in Hz.
+        If None, it will be determined from other parameters.
+        Default: None.
```

Comment:
Where happens the determination of the peak_resolution from other parameters, if it was 'None'?

Reply:
When the ```peak_resolution``` is ```None```, its value does not get derived from the other parameters since it is not needed for the subsequent steps. Only if ```peak_resolution``` is assigned an appropriate value, it is used to determine the time bandwidth product which in turn is used to computed ```num_tapers```, which controls the peak resolution of the estimate.

Reply:
Then `If None, it will be determined from other parameters.` is quite misleading since it's ambiguous what 'it' refers to